### PR TITLE
Refactor and clean up e2e framework utils

### PR DESCRIPTION
**Pull request by WanLinghao**
/kind cleanup
It is part of issue https://github.com/kubernetes/kubernetes/issues/76206. This patch handles test/e2e/framework/pv_util.go file. This file is more than 1000 lines, I split them into two directories based on their functions.
```release-note
NONE
```

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
@WanLinghao thanks for the PR but we are looking for smaller cleanups.
please focus on breaking up your work into smaller PRs.

you can still rename this one and make the commit touch only a small subset of files.

/hold

**Review by neolit123 — COMMENTED** [review]
@WanLinghao thanks for the PR but we are looking for smaller cleanups.
please focus on breaking up your work into smaller PRs.

you can still rename this one and make the commit touch only a small subset of files.

/hold

**Comment by alejandrox1** [comment]
Hi @WanLinghao any updates on this?

**Comment by josiahbjorgaard** [comment]
Hello! I'm a bug triage shadow for the 1.16 release cycle and considering this PR has not been updated for a long time, I'd like to check it's status. The code freeze is starting 29th August (about 2 weeks from now) and while there is plenty of time, we want to ensure that each PR has a chance to be merged on time.

As the PR is tagged for 1.16, is it still planned for this release?

**Comment by WanLinghao** [comment]
@josiahbjorgaard, I will update this very soon.

**Comment by WanLinghao** [comment]
@alejandrox1  Hi, sorry for  delay, it is ready to merge

File: test/e2e/framework/pv/pv.go
Comment (alejandrox1) [inline test/e2e/framework/pv/pv.go]: I think it would make sense to move `framework.PVReclaimingTimeout` into `/test/e2e/framework/pv` (e2epv) as well.
This one is used by e2epv and e2e/storage/volume_provisioning.go.

File: test/e2e/framework/pv/pv.go
Comment (alejandrox1) [inline test/e2e/framework/pv/pv.go]: I think it would make sense to move `framework.ClaimBindingTimeout` into `/test/e2e/framework/pv` (e2epv) as well.

File: test/e2e/framework/pv/pv.go
Comment (alejandrox1) [inline test/e2e/framework/pv/pv.go]: We could also move `framework.PVBindingTimeout` to `/test/e2e/framework/pv`.

File: test/e2e/framework/pv/pv.go
Comment (alejandrox1) [inline test/e2e/framework/pv/pv.go]: In the spirit of stripping away any PV-related functionality from the framework pkg (`/test/e2e/framework`).
I think it would be best to replace this call to `framework.WaitForPersistentVolumeClaimPhase` with the code being called by it.

https://github.com/kubernetes/kubernetes/blob/8b4fd4104df3374891e3a538eac4c4bdbd06719e/test/e2e/framework/util.go#L729-L731

`framework.WaitForPersistentVolumeClaimPhase` is just a wrapper for [WaitForPersistentVolumeClaimsPhase](https://github.com/kubernetes/kubernetes/blob/8b4fd4104df3374891e3a538eac4c4bdbd06719e/test/e2e/framework/util.go#L735).
We should move this method into `/test/e2e/framework/pv`.

**Review by alejandrox1 — COMMENTED** [review]
Left a couple comments of something we could strip out of the e2e test framework and move to e2epv but other than that this looks great!
Thank you for working on it @WanLinghao

**Comment by WanLinghao** [comment]
@alejandrox1 Comments addressed, I will squash them if you think everything is OK

**Comment by guineveresaenger** [comment]
Hi @WanLinghao and @alejandrox1 - I'm the Release Lead Shadow for 1.16 and given that it looks like potentially this is ready - could you go ahead and find an approver for this?

We need to merge this before we cut the rc release on Monday 9/9 so please let me know if you need some help in calling this out in sig channels!

**Comment by alejandrox1** [comment]
/assign @andrewsykim @timothysc 
ptal

**Review by timothysc — APPROVED** [review]
/lgtm 
/approve 

Thank you for doing this work!

**Comment by cblecker** [comment]
/lgtm cancel
needs rebase

**Comment by WanLinghao** [comment]
@cblecker rebase done, PTAL

**Review by alejandrox1 — APPROVED** [review]
/lgtm

**Comment by alejandrox1** [comment]
Thank you again for working on this!
Hopefully we can get it in before any more rebases are required :sweat_smile:

**Comment by WanLinghao** [comment]
@alejandrox1 rebase again..PTAL

**Comment by alejandrox1** [comment]
pref error: https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/76411/pull-kubernetes-integration/1174261957507682305/
/test pull-kubernetes-integration

**Review by alejandrox1 — APPROVED** [review]
/lgtm
again :sweat_smile:

**Pull request merged**
